app/frontend/src/stores/main-store.js

```diff
@@ -148,13 +154,18 @@ export class MainStore {
   }
 
   @action async renameGroup(groupId, newGroupName) {
-    await this._datastore.renameGroup(groupId, newGroupName)
-    this.setEditMode(EditMode.None)
+    if (newGroupName) {
+      await this._datastore.renameGroup(groupId, newGroupName)
+      this.setEditMode(EditMode.None)
+    }
   }
 
   @action async addEntityToGroup(groupId, entity) {
-    await this._datastore.addEntityToGroup(groupId, entity)
-    this.setEditMode(EditMode.None)
+    let validEthAddress = new RegExp('0[xX][0-9a-fA-F]+')
```

Comment:
Maybe it would be interesting to extract this regex into its own function since I'm sure it'll be used again for other fields.

```suggestion
   function validateEthAddress(address) {
    return address && new RegExp('0[xX][0-9a-fA-F]+').test(address)
  }
```